### PR TITLE
[flutter_tools] reland: drive service

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Overhaul of flutter drive in order to deliver a better experience, namely:

* flutter run and flutter drive now share more flags, so code paths that were previously only testable on run are now testable on drive.
* Removes web-initialize-platform as this is no longer used
* flutter drive correctly sets up a logger that shows native exceptions, by connecting to the vm service.
* VM service connection now provides access to memory info without launching devtools (only for debug/profile mode)
Web changes

Passes on the one test in the repo, otherwise the webdriver code has been isolated as much as possible

Additional NNBD related bug fixes:

No longer passes --enable-experiment to the test script. (FYI @blasten ). earlier we might have assumed that the flutter gallery benchmarks would be migrated along side the app and flutter driver, but only the app under test needs to be migrated. The test scripts should never be run with the experiment.

### CODE REVIEW

File: packages/flutter_tools/lib/src/drive/web_driver_service.dart [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      device,
+      target: mainPath,
+      buildInfo: buildInfo,
+      platform: globals.platform,
+    );
+    _residentRunner = webRunnerFactory.createWebRunner(
+      flutterDevice,
+      target: mainPath,
+      ipv6: ipv6,
+      debuggingOptions: buildInfo.isRelease ?
```
Comment (jonahwilliams) [inline packages/flutter_tools/lib/src/drive/web_driver_service.dart:59]: This is the fix. for mobile/desktop we default to start paused which does not yet work for the web driver code.

Comment (angjieli) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart:59]: How about other options? Is using the default value the right choice?

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/drive/web_driver_service.dart:59]: it is consistent with the previous behavior. I think supporting more flags is valuable but out of scope for this change

**Review by angjieli — APPROVED** [review]
LGTM

**Pull request merged**